### PR TITLE
Make I/O a lazy DAGNode in the coordinator

### LINKED ISSUE #645: Make I/O a lazy DAGNode in the coordinator
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by lidongze0629**
## Related issue number

Fixes #645 

- [x] `g.add_vertices()` and `g.add_edges()`
- [x]  `graphscope.load_from()`
- [x]  `context.output()`
- [x]  Add `context.output()` CI test
- [x]  Test load from OSS

### CODE REVIEW

File: coordinator/gscoordinator/coordinator.py [hunk 1]
```diff
@@ -819,6 +866,101 @@ def _fetch_gremlin_result(self, op: op_def_pb2.OpDef):
                 code=error_codes_pb2.OK, key=op.key, result=pickle.dumps(rlt)
             )
 
+    def _output(self, op: op_def_pb2.OpDef):
+        storage_options = pickle.loads(op.attr[types_pb2.STORAGE_OPTIONS].s)
```
Comment (sighingnow) [inline coordinator/gscoordinator/coordinator.py:870]: Passing the configuration using `json.loads/dumps` instead.

File: coordinator/gscoordinator/coordinator.py [hunk 2]
```diff
@@ -819,6 +866,101 @@ def _fetch_gremlin_result(self, op: op_def_pb2.OpDef):
...
+        fd = op.attr[types_pb2.FD].s.decode()
+        df = op.attr[types_pb2.VINEYARD_ID].s.decode()
+        engine_config = self._get_engine_config()
+        vineyard_endpoint = engine_config["vineyard_rpc_endpoint"]
+        vineyard_ipc_socket = engine_config["vineyard_socket"]
+        if self._launcher_type == types_pb2.K8S:
```
Comment (sighingnow) [inline coordinator/gscoordinator/coordinator.py:876]: Is the value of `self._launcher_type` also k8s?

**Review by sighingnow — APPROVED** [review]
LGTM.

File: coordinator/gscoordinator/coordinator.py [hunk 3]
```diff
@@ -48,8 +48,11 @@
 # capture system stdout
 sys.stdout = StdoutWrapper(sys.stdout)
 
+import vineyard
+import vineyard.io
```
Comment (sighingnow) [inline coordinator/gscoordinator/coordinator.py:52]: Put those two `import` into the function itself.

**Review by sighingnow — COMMENTED** [review]
Drop the `vineyard` and `vineyard-io` dependency from `graphscope` as well, and add them to `gscoordinator`.

File: coordinator/gscoordinator/coordinator.py (hunk 1 above)
Comment (lidongze0629) [reply coordinator/gscoordinator/coordinator.py:870]: done

File: coordinator/gscoordinator/coordinator.py (hunk 3 above)
Comment (lidongze0629) [reply coordinator/gscoordinator/coordinator.py:52]: done

**Comment by lidongze0629** [comment]
> Drop the `vineyard` and `vineyard-io` dependency from `graphscope` as well, and add them to `gscoordinator`.



> Drop the `vineyard` and `vineyard-io` dependency from `graphscope` as well, and add them to `gscoordinator`.

Add `vineyard` and `vineyard-io` as dependencies in `gscoordinator`, and will drop `vineyard-io` from the client after migrate `graph.load_from` and `graph.save_to`implementation into coordinator

**Pull request merged**